lib/assessment.js

```diff
@@ -32,6 +38,16 @@ var any = function any(array, callback, thisArg) {
 // made up of the array elements for which invoking
 // the callback returns a truthy value
 var select = function select(array, callback, thisArg) {
+  var result = true;
+var newArray = [];
+  array.forEach(function(currentValue, index, array) {
+
+    if(result = callback(currentValue, index, array)){
```

Comment:
This `if` statement's condition is an assignment statement. Fortunately, the variable you're assigning to is actually not important in this function, and the value an assignment statement evaluates to is whatever is on the right side of the assignment operator (`=`). So line 45 here is, aside from the side effect of assigning to `result`, equivalent to `if(callback(currentValue, index, array)) {`. This is good, because it means you're testing for the right thing.